### PR TITLE
[codex] Fix auto-mode session-transition abort handling

### LINKED ISSUE #5632: Fix auto-mode session-transition abort misclassification
## Problem

Auto-mode can stop after a successful Unit when Claude Code emits a late stream-aborted marker during session transition cleanup. In the observed `testingapp1` M003/S04 run, `execute-task` completed, `complete-slice` was dispatched, and then a late `Claude Code stream aborted by caller` event was classified as an active Unit abort, dropping the user back to the TUI.

## Root cause

`newSession()` intentionally aborts an in-flight previous stream during session transition, but that abort was emitted without causal ownership. GSD auto-mode could not distinguish an internal transition abort from a user/provider abort.

## Proposed fix

Add causal metadata to agent lifecycle events (`sessionId`, `turnId`, `abortOrigin`), tag session-transition aborts explicitly, preserve user abort tagging for public abort entry points, and teach the GSD auto Unit event classifier to ignore internal transition aborts while keeping real user/provider aborts observable.

## Validation

The patch should include regression coverage for the session-switch abort path and run the PR verification suite.

### PR DESCRIPTION
**Pull request by jeremymcs**
## TL;DR

**What:** Tags agent aborts with causal metadata and teaches GSD auto-mode to ignore internal session-transition aborts.
**Why:** Auto-mode could stop after a successful Unit when Claude Code emitted a late transition-cleanup abort marker.
**How:** Propagate `sessionId`, `turnId`, and `abortOrigin` through core agent events, tag session-switch/user abort paths, and classify transition aborts before Unit settlement.

## What

- Adds `AgentAbortOrigin` plus optional `sessionId`, `turnId`, and `abortOrigin` fields to agent and extension lifecycle events.
- Tags `newSession()`/session-transition aborts as `session-transition` and public TUI/RPC aborts as `user`.
- Adds a GSD auto Unit event classifier that ignores `session-transition` `agent_end` events for the active Unit.
- Keeps the narrow Claude Code `stream aborted by caller` placeholder guard as compatibility fallback for untyped/older events.
- Adds regression coverage for the observed session-switch abort misclassification path.

## Why

Closes #5632.

In the `testingapp1` M003/S04 auto-mode run, the Unit completed successfully and auto-mode dispatched `complete-slice`, but a late Claude Code abort placeholder from session transition cleanup arrived afterward. Because abort events had no ownership metadata, GSD treated that internal cleanup event as the active Unit aborting and dropped the user back to the TUI.

## How

The agent core now stamps lifecycle events with a per-turn id and current session id. `Agent.abort()` accepts a causal origin, `AgentSession` forwards that origin through `agent_end` and `stop`, and GSD auto-mode classifies `abortOrigin: "session-transition"` as internal cleanup instead of Unit failure. User abort entry points continue to flow through the normal cancellation path as `abortOrigin: "user"`.

## Validation

- `npm run verify:pr`
- Focused regression: `node --import ./src/resources/extensions/gsd/tests/resolve-ts.mjs --experimental-strip-types --test src/resources/extensions/gsd/tests/session-switch-abort-misclassification.test.ts`
- Focused lifecycle test: `node --import ./src/resources/extensions/gsd/tests/resolve-ts.mjs --experimental-strip-types --test packages/pi-coding-agent/src/core/agent-session-abort-order.test.ts`

## Change Type

- [x] `fix` — Bug fix
- [ ] `feat` — New feature or capability
- [ ] `refactor` — Code restructuring (no behavior change)
- [x] `test` — Adding or updating tests
- [ ] `docs` — Documentation only
- [ ] `chore` — Build, CI, or tooling changes

## Notes

AI-assisted PR. No co-author metadata added to the commit.

## Summary by CodeRabbit

* **New Features**
  * Agent lifecycle, turn, message and tool events may include optional sessionId and turnId; agent_end includes abortOrigin for clearer termination context.
  * Abort flows now propagate an abort origin through the runtime and extension hooks.

* **Bug Fixes**
  * Prevents treating internal session-transition aborts as user cancellations; hardened handling of stream-abort/message edge cases during unit recovery.

* **Tests**
  * Added regression tests for session-switch abort classification.

* **Documentation**
  * Updated docs and API reference to describe event metadata and abortOrigin semantics.

[Review Change Stack](https://app.coderabbit.ai/change-stack/gsd-build/gsd-2/pull/5633)

### CODE REVIEW

File: src/resources/extensions/gsd/bootstrap/agent-end-recovery.ts [hunk 1]
```diff
@@ -270,7 +287,14 @@ export async function handleAgentEnd(
     return;
   }
 
-  if (lastMsg && "stopReason" in lastMsg && lastMsg.stopReason === "aborted") {
+  if (isBareClaudeCodeStreamAbortPlaceholder(lastMsg)) {
+    // The Claude Code adapter can emit this placeholder after a prior turn has
+    // already completed and the next unit is active. It has no user/provider
+    // diagnostic value and must not cancel the newly-dispatched unit.
+    return;
+  }
```
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/bootstrap/agent-end-recovery.ts:295]: ⏳ **Accepted** — this comment requires code changes. Queuing fix...
🧰 **Agent running** — `codex` is working on the fix...

**Comment by jeremymcs** [comment]
⏳ **Accepted** — this comment requires code changes. Queuing fix...
🧰 **Agent running** — `codex` is working on the fix...

**Comment by jeremymcs** [comment]
🤖 **oh-my-pr** dispatched `codex` with the following prompt:

<details>
<summary>Agent prompt (click to expand)</summary>

````text
You are acting as an autonomous PR babysitter for gsd-build/gsd-2 PR #5633.
PR URL: https://github.com/gsd-build/gsd-2/pull/5633
Base repository: gsd-build/gsd-2
Head repository: gsd-build/gsd-2
Head branch: fix/auto-session-transition-abort-origin
Head remote: origin
You are running inside an isolated app-owned worktree under ~/.oh-my-pr.
Make only targeted changes that resolve the approved tasks.
Do not wait for user input, confirmation, or approval at any point.
Do not rewrite unrelated files.
Use the available git tooling for inspection and verification only.
If dependencies are missing, install them using the repository's lockfile/package manager as needed inside this isolated worktree.
Leave file edits uncommitted; the babysitter app will handle Git finalization after your run.
GitHub follow-up replies and review-thread resolution will be handled by the babysitter after your run.
If a task is invalid after inspection, explain it in your final response and include the exact audit token.

Approved review-comment tasks:
1. [review_comment] coderabbitai[bot]
   file=src/resources/extensions/gsd/bootstrap/agent-end-recovery.ts line=295
   sourceId=3213428001 sourceUrl=https://github.com/gsd-build/gsd-2/pull/5633#discussion_r3213428001
   replyKind=review_thread threadId=PRRT_kwDORj5hb86A1SlW threadResolved=false
   auditToken=codefactory-feedback:gh-review-comment-3213428001
   _⚠️ Potential issue_ | _🟠 Major_ | _⚡ Quick win_

**Don't swallow typed user aborts behind the Claude placeholder fallback.**

This early return ignores the event purely from the message text. If Claude Code emits the same placeholder on a real typed user cancel, `handleAgentEnd()` will now drop that cancel and keep auto-mode running, which conflicts with the PR goal to preserve real user/provider aborts. Gate this fallback to untyped or `"unknown"` origins only.


<details>
<summary>Suggested fix</summary>

```diff
-  if (isBareClaudeCodeStreamAbortPlaceholder(lastMsg)) {
+  if (
+    (event.abortOrigin == null || event.abortOrigin === "unknown") &&
+    isBareClaudeCodeStreamAbortPlaceholder(lastMsg)
+  ) {
     // The Claude Code adapter can emit this placeholder after a prior turn has
     // already completed and the next unit is active. It has no user/provider
     // diagnostic value and must not cancel the newly-dispatched unit.
     return;
   }
```
</details>

<!-- suggestion_start -->

<details>
<summary>📝 Committable suggestion</summary>

> ‼️ **IMPORTANT**
> Carefully review the code before committing. Ensure that it accurately replaces the highlighted code, contains no missing lines, and has no issues with indentation. Thoroughly test & benchmark the code to ensure it meets the requirements.

```suggestion
  if (
    (event.abortOrigin == null || event.abortOrigin === "unknown") &&
    isBareClaudeCodeStreamAbortPlaceholder(lastMsg)
  ) {
    // The Claude Code adapter can emit this placeholder after a prior turn has
    // already completed and the next unit is active. It has no user/provider
    // diagnostic value and must not cancel the newly-dispatched unit.
    return;
  }
```

</details>

<!-- suggestion_end -->

<details>
<summary>🤖 Prompt for AI Agents</summary>

```
Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@src/resources/extensions/gsd/bootstrap/agent-end-recovery.ts` around lines
290 - 295, The early return that ignores events when
isBareClaudeCodeStreamAbortPlaceholder(lastMsg) is true must be gated by the
message origin so real typed user/provider aborts aren't dropped: update the
conditional in the agent-end-recovery flow (where
isBareClaudeCodeStreamAbortPlaceholder and lastMsg are checked before calling
handleAgentEnd) to only return when the event origin is missing/unknown/untyped
(e.g., origin === undefined or origin === "unknown" or an explicit untyped
flag), and allow events with explicit typed user/provider origins to fall
through so handleAgentEnd() can process real cancels.
```

</details>

<!-- fingerprinting:phantom:medusa:grasshopper -->

<!-- 4e71b3a2 -->

<!-- This is an auto-generated comment by CodeRabbit -->
2. [review] coderabbitai[bot]
   file=n/a line=n/a
   sourceId=4258047726 sourceUrl=https://github.com/gsd-build/gsd-2/pull/5633#pullrequestreview-4258047726
   replyKind=review threadId=n/a threadResolved=n/a
   auditToken=codefactory-feedback:gh-review-4258047726
   **Actionable comments posted: 2**

> [!CAUTION]
> Some comments are outside the diff and can’t be posted inline due to platform limitations.
> 
> 
> 
> <details>
> <summary>⚠️ Outside diff range comments (2)</summary><blockquote>
> 
> <details>
> <summary>packages/pi-coding-agent/src/core/agent-session.ts (2)</summary><blockquote>
> 
> `698-769`: _⚠️ Potential issue_ | _🟠 Major_ | _⚡ Quick win_
> 
> **Forward `sessionId`/`turnId` on extension tool-execution events too.**
> 
> This method now preserves causal metadata for agent/turn/message events, but `tool_execution_*` still strips it when crossing into the extension layer. Extensions that need to correlate tool activity with the active turn/session still can't do that.
> 
>  
> <details>
> <summary>Suggested direction</summary>
> 
> ```diff
> 		} else if (event.type === "tool_execution_start") {
> 			const extensionEvent: ToolExecutionStartEvent = {
> 				type: "tool_execution_start",
> 				toolCallId: event.toolCallId,
> 				toolName: event.toolName,
> 				args: event.args,
> +				sessionId: event.sessionId,
> +				turnId: event.turnId,
> 			};
> 			await extensionRunner.emit(extensionEvent);
> 		} else if (event.type === "tool_execution_update") {
> 			const extensionEvent: ToolExecutionUpdateEvent = {
> 				type: "tool_execution_update",
> 				toolCallId: event.toolCallId,
> 				toolName: event.toolName,
> 				args: event.args,
> 				partialResult: event.partialResult,
> +				sessionId: event.sessionId,
> +				turnId: event.turnId,
> 			};
> 			await extensionRunner.emit(extensionEvent);
> 		} else if (event.type === "tool_execution_end") {
> 			const extensionEvent: ToolExecutionEndEvent = {
> 				type: "tool_execution_end",
> 				toolCallId: event.toolCallId,
> 				toolName: event.toolName,
> 				result: event.result,
> 				isError: event.isError,
> +				sessionId: event.sessionId,
> +				turnId: event.turnId,
> 			};
> 			await extensionRunner.emit(extensionEvent);
> 		}
> ```
> 
> Also extend the corresponding `ToolExecution*Event` interfaces in `packages/pi-coding-agent/src/core/extensions/types.ts`.
> </details>
> 
> <details>
> <summary>🤖 Prompt for AI Agents</summary>
> 
> ```
> Verify each finding against current code. Fix only still-valid issues, skip the
> rest with a brief reason, keep changes minimal, and validate.
> 
> In `@packages/pi-coding-agent/src/core/agent-session.ts` around lines 698 - 769,
> The tool execution extension events currently omit causal metadata; update the
> construction of ToolExecutionStartEvent, ToolExecutionUpdateEvent, and
> ToolExecutionEndEvent in agent-session.ts to include sessionId and turnId copied
> from the incoming event (e.g., when handling event.type ===
> "tool_execution_start"/"tool_execution_update"/"tool_execution_end" pass
> event.sessionId and event.turnId into the extensionEvent objects before calling
> extensionRunner.emit), and also extend the corresponding
> ToolExecutionStartEvent/ToolExecutionUpdateEvent/ToolExecutionEndEvent
> interfaces in packages/pi-coding-agent/src/core/extensions/types.ts to declare
> optional or required sessionId and turnId so extensions can correlate tool
> activity with the active session/turn.
> ```
> 
> </details>
> 
> ---
> 
> `1639-1659`: _⚠️ Potential issue_ | _🟠 Major_ | _⚡ Quick win_
> 
> **Keep `turnId` on the synthetic abort fallback events.**
> 
> This fallback is the exact path used when an abort interrupts the normal lifecycle, but the emitted `agent_end`/`stop` only carry `sessionId` and `abortOrigin`. Consumers still lose the active-turn attribution in that edge case.
> 
> <details>
> <summary>🤖 Prompt for AI Agents</summary>
> 
> ```
> Verify each finding against current code. Fix only still-valid issues, skip the
> rest with a brief reason, keep changes minimal, and validate.
> 
> In `@packages/pi-coding-agent/src/core/agent-session.ts` around lines 1639 - 1659,
> The synthetic abort fallback currently emits agent_end and emitStop without the
> active turn identifier; update the two emission calls (_extensionRunner.emit and
> _extensionRunner.emitStop) to include the active turnId so consumers retain turn
> attribution on aborts (add a turnId field using the session's active turn
> identifier, e.g. this.turnId or this.activeTurnId, into the agent_end payload
> and the emitStop payload alongside sessionId and abortOrigin; keep lastMessage
> and reason as-is).
> ```
> 
> </details>
> 
> </blockquote></details>
> 
> </blockquote></details>

<details>
<summary>🧹 Nitpick comments (1)</summary><blockquote>

<details>
<summary>packages/pi-coding-agent/src/core/agent-session-abort-order.test.ts (1)</summary><blockquote>

`242-257`: _⚡ Quick win_

**Also assert `abortOrigin` on `emitStop()`.**

This regression only records origin from `agent_end`, so a break in the parallel `emitStop(...abortOrigin)` path would still pass even though this PR threads origin through both lifecycle events.


<details>
<summary>Suggested test extension</summary>

```diff
 		const observedProcessingStates: boolean[] = [];
 		const observedOrigins: unknown[] = [];
+		const observedStopOrigins: unknown[] = [];

 		(session as any).agent.abort = () => {};
 		(session as any).agent.waitForIdle = async () => {};
 		(session as any)._extensionRunner = {
 			emit: async (event: any) => {
 				if (event.type === "agent_end") {
 					observedProcessingStates.push((session as any)._processingAgentEnd);
 					observedOrigins.push(event.abortOrigin);
 				}
 			},
-			emitStop: async () => {
+			emitStop: async (event: any) => {
 				observedProcessingStates.push((session as any)._processingAgentEnd);
+				observedStopOrigins.push(event.abortOrigin);
 			},
 		};

 		await session.abort({ origin: "session-transition" });

 		assert.deepEqual(observedProcessingStates, [true, true]);
 		assert.deepEqual(observedOrigins, ["session-transition"]);
+		assert.deepEqual(observedStopOrigins, ["session-transition"]);
 		assert.equal((session as any)._processingAgentEnd, false);
```
</details>

<details>
<summary>🤖 Prompt for AI Agents</summary>

```
Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@packages/pi-coding-agent/src/core/agent-session-abort-order.test.ts` around
lines 242 - 257, Test currently only records abortOrigin when
_extensionRunner.emit handles event.type === "agent_end", so emitStop's
abortOrigin is not asserted; update the test around (session as
any)._extensionRunner to have emitStop also capture event origin (or its passed
abortOrigin) and then after await session.abort({ origin: "session-transition"
}) assert that observedOrigins includes the abortOrigin from emitStop as well
(e.g., expect observedOrigins to contain "session-transition" twice or add a
separate assertion for the emitStop path). Refer to _extensionRunner.emit,
_extensionRunner.emitStop, abortOrigin, session.abort, observedOrigins and
observedProcessingStates when making the change.
```

</details>

</blockquote></details>

</blockquote></details>

<details>
<summary>🤖 Prompt for all review comments with AI agents</summary>

```
Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

Inline comments:
In `@packages/pi-agent-core/src/agent.ts`:
- Around line 490-491: The code currently creates a single turnId via
randomUUID() at the start of _runLoop(), which stamps the same UUID across
multiple logical turns; move creation of the UUID so a new randomUUID() is
generated each time a turn is emitted (i.e., where you emit
"turn_start"/"turn_end" or create a new logical turn in methods that handle
follow-ups/steering). Update all references that previously used the
_runLoop-scoped turnId (see usages around the turn emission code and the blocks
mentioned: the initial turnId at the top of _runLoop, the sections around lines
548-550 and 641-655) to accept a per-turn turnId parameter or local variable
created immediately before emitting the turn events, and ensure any functions
that propagate turn metadata (follow-up handlers, steering paths) receive and/or
generate a fresh turnId for each new logical turn.

In `@src/resources/extensions/gsd/bootstrap/agent-end-recovery.ts`:
- Around line 290-295: The early return that ignores events when
isBareClaudeCodeStreamAbortPlaceholder(lastMsg) is true must be gated by the
message origin so real typed user/provider aborts aren't dropped: update the
conditional in the agent-end-recovery flow (where
isBareClaudeCodeStreamAbortPlaceholder and lastMsg are checked before calling
handleAgentEnd) to only return when the event origin is missing/unknown/untyped
(e.g., origin === undefined or origin === "unknown" or an explicit untyped
flag), and allow events with explicit typed user/provider origins to fall
through so handleAgentEnd() can process real cancels.

---

Outside diff comments:
In `@packages/pi-coding-agent/src/core/agent-session.ts`:
- Around line 698-769: The tool execution extension events currently omit causal
metadata; update the construction of ToolExecutionStartEvent,
ToolExecutionUpdateEvent, and ToolExecutionEndEvent in agent-session.ts to
include sessionId and turnId copied from the incoming event (e.g., when handling
event.type ===
"tool_execution_start"/"tool_execution_update"/"tool_execution_end" pass
event.sessionId and event.turnId into the extensionEvent objects before calling
extensionRunner.emit), and also extend the corresponding
ToolExecutionStartEvent/ToolExecutionUpdateEvent/ToolExecutionEndEvent
interfaces in packages/pi-coding-agent/src/core/extensions/types.ts to declare
optional or required sessionId and turnId so extensions can correlate tool
activity with the active session/turn.
- Around line 1639-1659: The synthetic abort fallback currently emits agent_end
and emitStop without the active turn identifier; update the two emission calls
(_extensionRunner.emit and _extensionRunner.emitStop) to include the active
turnId so consumers retain turn attribution on aborts (add a turnId field using
the session's active turn identifier, e.g. this.turnId or this.activeTurnId,
into the agent_end payload and the emitStop payload alongside sessionId and
abortOrigin; keep lastMessage and reason as-is).

---

Nitpick comments:
In `@packages/pi-coding-agent/src/core/agent-session-abort-order.test.ts`:
- Around line 242-257: Test currently only records abortOrigin when
_extensionRunner.emit handles event.type === "agent_end", so emitStop's
abortOrigin is not asserted; update the test around (session as
any)._extensionRunner to have emitStop also capture event origin (or its passed
abortOrigin) and then after await session.abort({ origin: "session-transition"
}) assert that observedOrigins includes the abortOrigin from emitStop as well
(e.g., expect observedOrigins to contain "session-transition" twice or add a
separate assertion for the emitStop path). Refer to _extensionRunner.emit,
_extensionRunner.emitStop, abortOrigin, session.abort, observedOrigins and
observedProcessingStates when making the change.
```

</details>

<details>
<summary>🪄 Autofix (Beta)</summary>

Fix all unresolved CodeRabbit comments on this PR:

- [ ] <!-- {"checkboxId": "4b0d0e0a-96d7-4f10-b296-3a18ea78f0b9"} --> Push a commit to this branch (recommended)
- [ ] <!-- {"checkboxId": "ff5b1114-7d8c-49e6-8ac1-43f82af23a33"} --> Create a new PR with the fixes

</details>

---

<details>
<summary>ℹ️ Review info</summary>

<details>
<summary>⚙️ Run configuration</summary>

**Configuration used**: defaults

**Review profile**: CHILL

**Plan**: Pro Plus

**Run ID**: `0156ef81-5117-4f4b-87c8-2274642a2641`

</details>

<details>
<summary>📥 Commits</summary>

Reviewing files that changed from the base of the PR and between 4edec462062bbcc8559b2af4756ffaf117441a52 and 0927055bcdf0e989de6d93a1f389e704cb2e426f.

</details>

<details>
<summary>📒 Files selected for processing (11)</summary>

* `packages/pi-agent-core/src/agent.ts`
* `packages/pi-agent-core/src/types.ts`
* `packages/pi-coding-agent/src/core/agent-session-abort-order.test.ts`
* `packages/pi-coding-agent/src/core/agent-session.ts`
* `packages/pi-coding-agent/src/core/extensions/types.ts`
* `packages/pi-coding-agent/src/modes/interactive/interactive-mode.ts`
* `packages/pi-coding-agent/src/modes/rpc/rpc-mode.ts`
* `src/resources/extensions/gsd/auto/types.ts`
* `src/resources/extensions/gsd/auto/unit-runner-events.ts`
* `src/resources/extensions/gsd/bootstrap/agent-end-recovery.ts`
* `src/resources/extensions/gsd/tests/session-switch-abort-misclassification.test.ts`

</details>

</details>

<!-- This is an auto-generated comment by CodeRabbit for review status -->

Approved status-check tasks:
None

Approved documentation tasks:
Documentation updates are required for this PR.
Assessment summary: The PR changes exported agent/extension event contracts by adding sessionId, turnId, and abortOrigin plus abort origin semantics. Extension/API docs should document these optional fields and the new abort origin values so consumers know how to handle session-transition aborts.
Update the appropriate repository documentation for these changes.
Choose the right docs files for this repository (for example README, docs pages, API/config/operator docs).
If, after inspection, the repository documentation is already accurate or there is no appropriate docs target, leave docs unchanged and report that using the docs summary block with `no_change`.

When done:
1) Run the relevant verification for your changes.
2) Leave any changed files in the worktree for the babysitter app to finalize.
3) For each feedback item you addressed or were blocked on, emit a summary block in the following format:
   FEEDBACK_SUMMARY_START <auditToken>
   <A concise 1-2 sentence summary of what you did or why you were blocked>
   FEEDBACK_SUMMARY_END
   Include one block per audit token. These summaries will be posted as follow-up comments on the PR.
4) If documentation tasks were assigned, emit exactly one docs summary block in the following format:
   DOCS_SUMMARY_START <changed|no_change>
   <A concise 1-2 sentence summary of the docs you updated, or why no docs changes were necessary after inspection>
   DOCS_SUMMARY_END
````

</details>

**Comment by jeremymcs** [comment]
🤖 Dispatched `codex` with the following prompt:

<details>
<summary>Agent prompt (click to expand)</summary>

```text
You are acting as an autonomous PR babysitter for gsd-build/gsd-2 PR #5633.
PR URL: https://github.com/gsd-build/gsd-2/pull/5633
Base repository: gsd-build/gsd-2
Head repository: gsd-build/gsd-2
Head branch: fix/auto-session-transition-abort-origin
Head remote: origin
You are running inside an isolated app-owned worktree under ~/.oh-my-pr.
Make only targeted changes that resolve the approved tasks.
Do not wait for user input, confirmation, or approval at any point.
Do not rewrite unrelated files.
Use the available git tooling for inspection and verification only.
If dependencies are missing, install them using the repository's lockfile/package manager as needed inside this isolated worktree.
Leave file edits uncommitted; the babysitter app will handle Git finalization after your run.
GitHub follow-up replies and review-thread resolution will be handled by the babysitter after your run.
If a task is invalid after inspection, explain it in your final response and include the exact audit token.

Approved review-comment tasks:
None

Approved status-check tasks:
None

Approved documentation tasks:
Documentation updates are required for this PR.
Assessment summary: The PR changes exported agent/extension event contracts by adding sessionId, turnId, and abortOrigin plus abort origin semantics. Extension/API docs should document these optional fields and the new abort origin values so consumers know how to handle session-transition aborts.
Update the appropriate repository documentation for these changes.
Choose the right docs files for this repository (for example README, docs pages, API/config/operator docs).
If, after inspection, the repository documentation is already accurate or there is no appropriate docs target, leave docs unchanged and report that using the docs summary block with `no_change`.

When done:
1) Run the relevant verification for your changes.
2) Leave any changed files in the worktree for the babysitter app to finalize.
3) For each feedback item you addressed or were blocked on, emit a summary block in the following format:
   FEEDBACK_SUMMARY_START <auditToken>
   <A concise 1-2 sentence summary of what you did or why you were blocked>
   FEEDBACK_SUMMARY_END
   Include one block per audit token. These summaries will be posted as follow-up comments on the PR.
4) If documentation tasks were assigned, emit exactly one docs summary block in the following format:
   DOCS_SUMMARY_START <changed|no_change>
   <A concise 1-2 sentence summary of the docs you updated, or why no docs changes were necessary after inspection>
   DOCS_SUMMARY_END
```

</details>

**Comment by jeremymcs** [comment]
🤖 Dispatched `codex` with the following prompt:

<details>
<summary>Agent prompt (click to expand)</summary>

```text
You are acting as an autonomous PR babysitter for gsd-build/gsd-2 PR #5633.
PR URL: https://github.com/gsd-build/gsd-2/pull/5633
Base repository: gsd-build/gsd-2
Head repository: gsd-build/gsd-2
Head branch: fix/auto-session-transition-abort-origin
Head remote: origin
You are running inside an isolated app-owned worktree under ~/.oh-my-pr.
Make only targeted changes that resolve the approved tasks.
Do not wait for user input, confirmation, or approval at any point.
Do not rewrite unrelated files.
Use the available git tooling for inspection and verification only.
If dependencies are missing, install them using the repository's lockfile/package manager as needed inside this isolated worktree.
Leave file edits uncommitted; the babysitter app will handle Git finalization after your run.
GitHub follow-up replies and review-thread resolution will be handled by the babysitter after your run.
If a task is invalid after inspection, explain it in your final response and include the exact audit token.

Approved review-comment tasks:
None

Approved status-check tasks:
None

Approved documentation tasks:
Documentation updates are required for this PR.
Assessment summary: Docs were updated for sessionId/turnId and abortOrigin, but they omit tool_execution_start, tool_execution_update, and tool_execution_end from the documented metadata coverage even though the code now stamps those events with sessionId and turnId. Update the event metadata/API docs to include tool execution events or narrow the implementation/docs consistently.
Update the appropriate repository documentation for these changes.
Choose the right docs files for this repository (for example README, docs pages, API/config/operator docs).
If, after inspection, the repository documentation is already accurate or there is no appropriate docs target, leave docs unchanged and report that using the docs summary block with `no_change`.

When done:
1) Run the relevant verification for your changes.
2) Leave any changed files in the worktree for the babysitter app to finalize.
3) For each feedback item you addressed or were blocked on, emit a summary block in the following format:
   FEEDBACK_SUMMARY_START <auditToken>
   <A concise 1-2 sentence summary of what you did or why you were blocked>
   FEEDBACK_SUMMARY_END
   Include one block per audit token. These summaries will be posted as follow-up comments on the PR.
4) If documentation tasks were assigned, emit exactly one docs summary block in the following format:
   DOCS_SUMMARY_START <changed|no_change>
   <A concise 1-2 sentence summary of the docs you updated, or why no docs changes were necessary after inspection>
   DOCS_SUMMARY_END
```

</details>
